flask_resty/pagination.py

```diff
@@ -188,21 +188,52 @@ def spec_declaration(self, path, spec, **kwargs):
 class CursorPaginationBase(LimitPagination):
     cursor_arg = 'cursor'
 
+    def ensure_query_sorting(self, query, view):
+        sorting_field_orderings, missing_field_orderings = (
+            self.get_sorting_and_missing_field_orderings(view)
+        )
+
+        query = view.sorting.sort_query_by_fields(
+            query,
+            view,
+            missing_field_orderings,
+        )
+        field_orderings = sorting_field_orderings + missing_field_orderings
+
+        return query, field_orderings
+
     def get_field_orderings(self, view):
+        sorting_field_orderings, missing_field_orderings = (
+            self.get_sorting_and_missing_field_orderings(view)
+        )
+        return sorting_field_orderings + missing_field_orderings
+
+    def get_sorting_and_missing_field_orderings(self, view):
         sorting = view.sorting
         assert sorting is not None, (
             "sorting must be defined when using cursor pagination"
         )
 
-        field_orderings = sorting.get_request_field_orderings()
-        for id_field in view.id_fields:
-            assert (
-                id_field in (field_name for field_name, _ in field_orderings)
-            ), (
-                "ordering does not include {}".format(id_field)
-            )
+        sorting_field_orderings = sorting.get_request_field_orderings(view)
 
-        return field_orderings
+        sorting_ordering_fields = frozenset(
+            field_name for field_name, _ in sorting_field_orderings,
+        )
+
+        # For convenience, use the ascending setting on the last explicit
```

Comment:
why is this better than just setting it to `True` every time?

Reply:
if you do `-size`, it’d be nice to get the reverse